gandalf.js

```diff
@@ -1,131 +1,127 @@
 /**
```

Comment:
SCRIPT-I like how you've used promise chaining here, it makes the code flow very clear.